### PR TITLE
[DNM] jewel: ceph-disk does not support cluster names different than 'ceph'

### PR DESCRIPTION
**Pull request by shinobu-x**
http://tracker.ceph.com/issues/18972

### CODE REVIEW

**Comment by dachary** [comment]
http://pulpito.ceph.com/smithfarm-2017-02-20_21:30:56-ceph-disk-wip-jewel-backports-distro-basic-vps/ fails maybe because of this backport

**Comment by dachary** [comment]
Please cherry-pick from https://github.com/ceph/ceph/pull/13573/commits/7f66672b675abbc0262769d32a38112c781fefac instead. These two commits should have not be commited in master and must not be backported.

**Comment by shinobu-x** [comment]
@dachary Recommitted with suggested commit. Please check, if build is good.

File: src/ceph-disk/ceph_disk/main.py [hunk 1]
```diff
@@ -2340,6 +2351,11 @@ def populate(self):
         command_check_call(args)
         write_one_line(path, 'osd-uuid', self.args.osd_uuid)
         self.create_key()
```
Comment (dachary) [inline src/ceph-disk/ceph_disk/main.py]: you have create_key here and below. This conflict resolution was incorrect.

**Comment by dachary** [comment]
@shinobu-x when there is a conflict, please make sure you explain, in the commit message, what was the origin of the conflict and how your resolved it.

**Comment by dachary** [comment]
For the record the conflict comes from the fact that create_key() has an argument in master but not in jewel. This happened because of a bad revert/merge situation with 1c747473600a9a7f528ad71a889c35d85ac115b1.

**Comment by shinobu-x** [comment]
@dachary Thank you for your advice.

**Comment by shinobu-x** [comment]
```
root@octopus:ceph [wip-17821-jewel]$ git diff c9ece04 7376878 src/ceph-disk/ceph_disk/main.py
diff --git a/src/ceph-disk/ceph_disk/main.py b/src/ceph-disk/ceph_disk/main.py
index ffe0311..9ba19ad 100755
--- a/src/ceph-disk/ceph_disk/main.py
+++ b/src/ceph-disk/ceph_disk/main.py
```
...
```
-        self.create_key()`
+        if self.args.cluster_uuid is None:
+            self.args.cluster_uuid = get_fsid(cluster=self.args.cluster)
+        write_one_line(path, 'ceph_fsid', self.args.cluster_uuid)
+        self.create_key(self.args.cluster)
```

**Comment by shinobu-x** [comment]
Cherry-picked code from commit 7f66672 made conflict as below.
```
diff --cc src/ceph-disk/ceph_disk/main.py
index ffe0311,c9f0041..0000000
--- a/src/ceph-disk/ceph_disk/main.py
+++ b/src/ceph-disk/ceph_disk/main.py
@@@ -2339,7 -2552,10 +2350,14 @@@ class Lockbox(object)
          LOG.debug('Mounting lockbox ' + str(" ".join(args)))
          command_check_call(args)
          write_one_line(path, 'osd-uuid', self.args.osd_uuid)
++<<<<<<< HEAD
 +        self.create_key()
++=======
+         if self.args.cluster_uuid is None:
+             self.args.cluster_uuid = get_fsid(cluster=self.args.cluster)
+         write_one_line(path, 'ceph_fsid', self.args.cluster_uuid)
+         self.create_key(self.args.cluster)
++>>>>>>> 7f66672... ceph-disk: dmcrypt activate must use the same cluster as prepare
```
Because of context,  this won't work properly in jewel:
`+         self.create_key(self.args.cluster)`

Because of that, this was selected:
` +        self.create_key()`

And moved a selected line:
* from just before
`+         if self.args.cluster_uuid is None:`
* to just after:
`+         write_one_line(path, 'ceph_fsid', self.args.cluster_uuid)`

**Comment by shinobu-x** [comment]
retest this please

**Comment by shinobu-x** [comment]
retest this please

**Comment by shinobu-x** [comment]
@dachary When you get a chance, please re-reviewing this PR?

**Comment by shinobu-x** [comment]
retest this please

**Comment by smithfarm** [comment]
@shinobu-x You can easily edit the commit message using `git commit --amend` and then force push.

**Comment by shinobu-x** [comment]
@smithfarm Thanks for your quick reply. I will follow you and retest it out then let you know.

**Comment by shinobu-x** [comment]
retest this please

**Comment by shinobu-x** [comment]
@smithfarm Please have a look at this PR when you get a chance.

**Comment by shinobu-x** [comment]
retest this please

**Comment by shinobu-x** [comment]
retest this please

**Comment by smithfarm** [comment]
@shinobu-x This commit message is still not quite right. When you do `git cherry-pick -x $MASTER_SHA1` git prepares a commit message for you. It looks just like the original commit message except that there is a line appended like this: `(cherry picked from ...)` and after that some lines (commented out in newer git versions) like "Conflicts:" and the names of the files where the conflicts were found. Uncomment these lines and add your conflict resolution explanation to each file. 

For examples, you can look in the `jewel` branch like this:

```
git checkout jewel
git log --grep 'Conflicts:'
```

**Comment by smithfarm** [comment]
Closing in favor of #14765

**Pull request closed without merging**